From 56b41f9a62aa7cd911fee5dc73f927b6b82b67c9 Mon Sep 17 00:00:00 2001
From: George Kiagiadakis <george.kiagiadakis@collabora.co.uk>
Date: Tue, 16 Aug 2011 19:47:02 +0300
Subject: [PATCH 045/119] mpeg4videoparse: Use gst_bit_reader_skip to skip
 more than 32 bits.

GET_BITS is a macro for gst_bit_reader_get_bits_uint32, which cannot
read more than 32 bits and will fail in this case where it is called
to read 79 bits. Since we want to skip those bits, gst_bit_reader_skip
is more appropriate in this case.
---
 gst/mpeg4videoparse/mpeg4parse.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/gst/mpeg4videoparse/mpeg4parse.c b/gst/mpeg4videoparse/mpeg4parse.c
index 3cd6df8..d386ac9 100644
--- a/gst/mpeg4videoparse/mpeg4parse.c
+++ b/gst/mpeg4videoparse/mpeg4parse.c
@@ -138,7 +138,8 @@ gst_mpeg4_params_parse_vo (MPEG4Params * params, GstBitReader * br)
     GET_BITS (br, 1, &bits);
     if (bits) {
       /* skip vbv_parameters */
-      GET_BITS (br, 79, &bits);
+      if (!gst_bit_reader_skip (br, 79))
+        goto failed;
     }
   }
 
-- 
1.7.5.4

